go/worker/keymanager/worker.go

```diff
@@ -175,12 +176,15 @@ func (w *Worker) CallEnclave(ctx context.Context, data []byte) ([]byte, error) {
 	case getPublicKeyRequestMethod:
 		// Anyone can get public keys.
 	default:
-		// Defer to access control to check the policy.
-		w.RLock()
-		_, allowed := w.accessList[peerID]
-		w.RUnlock()
-		if !allowed {
-			return nil, fmt.Errorf("not authorized")
+		_, privatePeered := w.privatePeers[peerID]
```

Comment:
Could merge the assignment and if.

`if _, privatePeered := w.privatePeers[peerID]; !privatePeered {`